BrainPortal/app/models/tool_config.rb

```diff
@@ -337,7 +339,18 @@ def container_rules #:nodoc:
     if self.container_engine.present? && ( self.containerhub_image_name.blank? && self.container_image_userfile_id.blank? )
       errors[:container_engine] = "a container hub image name or a container image userfile ID should be set when the container engine is set"
     end
+
+    if self.container_engine.present? && self.container_engine == "Singularity" 
+      if self.container_index_location.present? && ! /^[a-z0-9]+\:\/\/$/.match(self.container_index_location)
+        errors[:container_index_location] = "Invalid index location type for container engine Singularity. Should end in '://'."
```

Comment:
The error messages are generated automatically using the attribute name as a subject for the sentence; so right now the error message will look weird; test it with the interface and read it carefully.